actor-typed-tests/src/test/scala/org/apache/pekko/actor/typed/SupervisionSpec.scala

```diff
@@ -401,40 +411,16 @@ class SupervisionSpec extends ScalaTestWithActorTestKit("""
         throw new Exc3()
         targetBehavior(probe.ref)
       })
-      val behv = Behaviors.supervise(failedSetup).onFailure[Throwable](SupervisorStrategy.stop)
+      val behv = supervise(failedSetup).onFailure[Throwable](SupervisorStrategy.stop)
       LoggingTestKit.error[Exc3].expect {
         spawn(behv)
       }
     }
 
-    "support nesting exceptions with different strategies" in {
+    def throwIOExceptionThenIllegalArgumentException(behvFactory: ActorRef[Event] => Behavior[Command]): Unit = {
```

Comment:
`"support nesting exceptions with outer restart and inner backoff strategies" in {`
and 
`"support nesting exceptions with different strategies" in {`

almost same, Therefore, it is compressed into the same method here